docker/runtime_env_container/Dockerfile

```diff
@@ -0,0 +1,12 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+COPY python/ray/tests/runtime_env_container/ /home/ray/tests/
+
+# Install podman
+RUN sudo apt-get update && \
+    sudo apt install software-properties-common uidmap -y && \
+    sudo sh -c "echo 'deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list" && \
+    sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 4D64390375060AA4 && \
+    sudo apt-get update && \
```

Comment:
could you only `apt-get update` once?

pressumably:

1. add deb source
2. apt update
3. apt install all packages you need

Reply:
So I tried removing the apt-get update at the beginning, and it can't seem to find the packages `software-properties-common` and `uidmap`, not sure why 😕

Reply:
first, always use `apt-get` not `apt` in docker files. `apt` is not for humanless scenarios.

then, the thing I am asking you to do is:

````
    sudo sh -c "echo 'deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list" && \
    sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 4D64390375060AA4 && \
    sudo apt-get update && \
    sudo apt-get install -y software-properties-common uidmap podman
````

the reason is that:

apt-get update takes some time and there is no reason to do it twice. you want to add a new deb source so that you can install `podman`, and that should be done before `apt-get update`

I actually do not quite understand why `software-properties-common` and `uidmap` are required here.

Reply:
@aslonnie for my own understanding, why is `apt-get` preferred over `apt` for humanless scenarios (and vice versa)?

Reply:
> why is apt-get preferred over apt for humanless scenarios (and vice versa)?

https://askubuntu.com/questions/990823/apt-gives-unstable-cli-interface-warning

APT shows the warning itself when being used on a non interactive terminal.

from `apt`'s manpage:

> The [apt](https://manpages.ubuntu.com/manpages/xenial/man8/apt.8.html)(8) commandline is designed as an end-user tool and it may change behavior between
       versions. While it tries not to break backward compatibility this is not guaranteed either
       if a change seems beneficial for interactive use.